### PR TITLE
Add in-repo documentation for user-selectable capabilities

### PR DESCRIPTION
**Pull request by kirankt**
Add in-repo documentation for user-selectable capabilities

### CODE REVIEW

File: docs/user/customization.md [hunk 1]
```diff
@@ -60,9 +59,15 @@ The following `install-config.yaml` properties are available:
...
     * `noProxy` (optional string): A comma-separated list of domains and [CIDRs][cidr-notation] for which the proxy should not be used.
+* `publish` (optional string): This controls how the user facing endpoints of the cluster like the Kubernetes API, OpenShift routes etc. are exposed.
+    Valid values are `External` (the default) and `Internal`.
 * `pullSecret` (required string): The secret to use when pulling images.
 * `sshKey` (optional string): The public Secure Shell (SSH) key to provide access to instances.
 
+### Capabilities
+
+* `baselineCapabilitySet` (optional string): Selects an initial set of optional capabilities to enable. Valid values are `vCurrent` (the default), `v4.11` and `None`.
+* `additionalEnabledCapabilities` (optional array of strings): Extends the set of managed capabilities beyond the baseline defined in `baselineCapabilitySet`. Default is an empty set. Valid values are `openshift-samples` and `baremetal`.
```
Comment (wking) [inline docs/user/customization.md:70]: You validate these, right?  Have you bumped your vendored openshift/api to pull in the `baremetal` addition?

**Comment by patrickdillon** [comment]
Example, please!

File: docs/user/customization.md (hunk 1 above)
Comment (wking) [reply docs/user/customization.md:70]: This line will need a bump for `marketplace` for #5755.  If #5755 lands first, then this PR would need the bump.  If this PR lands first, then #5755 would need the bump.  Or a third follow up PR could come in with the bump behind both this PR and #5755.

File: docs/user/customization.md [hunk 2]
```diff
@@ -60,9 +59,15 @@ The following `install-config.yaml` properties are available:
...
     * `httpsProxy` (optional string): The URL of the proxy for HTTPS requests.
     * `noProxy` (optional string): A comma-separated list of domains and [CIDRs][cidr-notation] for which the proxy should not be used.
+* `publish` (optional string): This controls how the user facing endpoints of the cluster like the Kubernetes API, OpenShift routes etc. are exposed.
+    Valid values are `External` (the default) and `Internal`.
 * `pullSecret` (required string): The secret to use when pulling images.
 * `sshKey` (optional string): The public Secure Shell (SSH) key to provide access to instances.
 
+### Capabilities
+
+* `baselineCapabilitySet` (optional string): Selects an initial set of optional capabilities to enable. Valid values are `vCurrent` (the default), `v4.11` and `None`.
```
[the remark this answers is not in the document]
Comment (kirankt) [reply docs/user/customization.md:69]: vCurrent has a list of baseline capabilities that get installed by default.

File: docs/user/customization.md (hunk 1 above)
Comment (kirankt) [reply docs/user/customization.md:70]: > This line will need a bump for `marketplace` for #5755. If #5755 lands first, then this PR would need the bump. If this PR lands first, then #5755 would need the bump. Or a third follow up PR could come in with the bump behind both this PR and #5755.

My guess is probably option 3. I'll send a follow up PR with new vendoring and an update to this doc, which should be the norm anyway going forward. Thanks.

File: docs/user/customization.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sadasu) [reply docs/user/customization.md:69]: Does the baseline capability 'vCurrent' include all capabilities thus providing the `additionalEnabledCapabilities` unnecessary?
I think it is worth mentioning if the default (vCurrent) value for `baselineCapabilitySet` already includes all optional capabilities or not.

[the remark this answers is not in the document]
Comment (sadasu) [reply docs/user/customization.md:69]: @wking, @kirankt  with the default settings:
`baselineCapabilitySet`: vCurrent
`additionalEnabledCapabilities`: None
would all optional components be installed?

If yes, would it be worth mentioning that in this document?

[the remark this answers is not in the document]
Comment (wking) [reply docs/user/customization.md:69]: [`vCurrent` docs][1]:

> ClusterVersionCapabilitySetCurrent is the recommended set of optional capabilities to enable for the cluster's current version of OpenShift.

We're currently recommending all three caps in `vCurrent`, but there is no commitment to continuing to include those caps or to always adding each new cap.

I'm not installer, so I have no stake in how much openshift/api context folks feel like maintaining here.  If it was me, I'd just point at the [floating][2] [docs][3].

[1]: https://github.com/openshift/api/blob/8107b7a38e530cedbd3e3af87ae93bca00e0fa84/config/v1/types_cluster_version.go#L265
[2]: https://pkg.go.dev/github.com/openshift/api/config/v1#ClusterVersionCapability
[3]: https://pkg.go.dev/github.com/openshift/api/config/v1#ClusterVersionCapabilitySet

**Comment by sadasu** [comment]
I think this is complete except for the reference to the https://pkg.go.dev/github.com/openshift/api/config/v1#ClusterVersionCapability as mentioned in https://github.com/openshift/installer/pull/5732#discussion_r838090567.

**Comment by kirankt** [comment]
/skip
This is a doc file update. We should skip CI.

**Pull request merged**